tests/Unit/Adapter/Product/Update/Filler/PropertyFillerTestCase.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace Tests\Unit\Adapter\Product\Update\Filler;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Product\Update\Filler\ProductUpdatablePropertyFillerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\ValueObject\NoManufacturerId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductCondition;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductVisibility;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint;
+use Product;
+
+abstract class PropertyFillerTestCase extends TestCase
+{
+    protected const DEFAULT_LANG_ID = 1;
+    protected const DEFAULT_SHOP_ID = 2;
+    protected const PRODUCT_ID = 3;
+
+    /**
+     * @dataProvider getDataForTestFillsUpdatableProperties
+     *
+     * @param UpdateProductCommand $command
+     * @param array<int|string, string|int[]> $expectedUpdatableProperties
+     * @param Product $expectedProduct
+     */
+    public function testFillsUpdatableProperties(
+        UpdateProductCommand $command,
+        array $expectedUpdatableProperties,
+        Product $expectedProduct
+    ): void {
+        $product = $this->mockDefaultProduct();
+
+        $this->assertSame(
+            $expectedUpdatableProperties,
+            $this->getFiller()->fillUpdatableProperties($product, $command)
+        );
+
+        $this->assertEquals($expectedProduct, $product);
+    }
+
+    abstract public function getDataForTestFillsUpdatableProperties(): iterable;
+
+    abstract public function getFiller(): ProductUpdatablePropertyFillerInterface;
```

Comment:
```suggestion
    abstract protected function getDataForTestFillsUpdatableProperties(): iterable;

    abstract protected function getFiller(): ProductUpdatablePropertyFillerInterface;
```

These two could and should be protected no? Maybe `getDataForTestFillsUpdatableProperties` is a bit different since a data provider so I understood we may be forced to keep it public But for `getFiller` it seems we don't need to have it public, or am I missing something?